packages/cli/src/active-executions.ts

```diff
@@ -170,6 +170,7 @@ export class ActiveExecutions {
 	finalizeExecution(executionId: string, fullRunData?: IRun) {
 		if (!this.has(executionId)) return;
 		const execution = this.getExecutionOrFail(executionId);
+		execution?.responsePromise?.resolve({});
```

Comment:
This should be removed, and we should call `resolveResponsePromise` from a relevant place in `webhook-helpers.ts` instead.

Reply:
@netroy 
yes, after more testing this is not fixes all issues, but I don't think we could do this from `webhook-helpers.ts`

we need to resolve `responsePromise`, if there is any, once, after workflow finishes, as a clean up, if some page waits for response

do you have some ideas how it could be done?

this [workflow](https://test-fru.stage-app.n8n.cloud/workflow/0J7HbnPWmvfCSBFM) demonstrate issue

Reply:
We don't need to block this change. 
I unfortunately don't have enough time or context to provide a solution here. I can try to fix this in another PR.

Reply:
just FYI: this change might end up calling the `responsePromise.resolve` multiple times. once from this line, and another from any code calling `resolveResponsePromise`